### ISSUE TITLE
Check open (undrafted) PR count per contributor and send a Slack notification prompting alternative tasks instead of new (reviewable) PRs that could be done

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse#2296]
## Problem

Openverse contributors tend to highly prioritise code contributions over other types of contributions. Part of this comes from the enthusiasm we have for the improvements we can make to the project. Part of this also comes, I believe, from a lack of visibility, information, and expectations for other types of tasks that can be worked on that do not necessarily result in a new immediate code contribution. Unfortunately, this also leads to some of these tasks either being completely ignored by the entire team or being unevenly distributed across the core maintainers, neither of which are good for the overall health of the project or individual maintainers.

## Description

Add a new workflow triggered on PR open to count the number of reviewable[^reviewable] PRs currently open from the contributor opening a new PR. If it equals or exceeds 3 (see below "why three") and the contributor is present in the GitHub ID to Slack ID mapping, send them a pre-formatted message along the following lines:

```md
Howdy <contributor>!

You currently have <x> number of PRs open with requested reviews (totalling <x * 2> required reviews). To help ease the review burden, increase review velocity for older PRs, and improve the equitable distribution of project maintenance tasks across the team, please consider reviewing [this list of alternative project tasks](<link to docs page>) instead of working on new code contributions.
```

The number of open reviews should be dynamic in case someone uses their discretion to open PRs beyond the baseline number. We agree that this should be allowed as some situations will call for it (critical issues that you can solve quickly, etc). However, the reminder should still go out. Including the number of requested reviews helps emphasise how significant the review burden is each time a new PR is opened, especially compared to the capacity that reviewers will have to accomplish their own work.

[^reviewable]: Open and un-drafted. Reviewers are only pinged to review PRs that are "ready for review", i.e., not drafts. We should be able to create draft PRs and have them ready to mark "ready for review" once our PRs in need of review are reviewed.

### Why three?

We have a limited number of reviewers available for each PR. Each PR requires 2 approvals. For simplicity's sake, let's assume a consistent pool of 7 available reviewers and regular code contributors for the project. If each person has 3 PRs up for review, that would be 24 PRs. That would mean each reviewer, at any given time, would have just about 7 PRs assigned for them to review at a time:

2 required reviewers * 24 PRs = 48 total required reviews
48 reviews / 7 available reviewers = 6.86 reviews per person

That already seems like a lot, but the reality is that not everyone will be opening so many PRs to review at a time. However, because the pool of reviewers is not evenly spread across the different stacks and because some stacks tend to have a higher number of smaller PRs (which can often nonetheless still take significant time to review), I think 3 is appropriate. In any case, this issue should be implemented in such a way that this number can be easily adjusted up or down depending on the experience of the team after some time.

## Potential future improvements

Weigh the number of reviewable PRs for each contributor against the stack labels for those PRs. Because our reviewers are distributed unevenly across the app, it could make sense to prompt people to "either do one of these maintenance tasks or consider picking up an issue in a different part of the stack".

**Comment by AetherUnbound** [comment]
This sounds great! +1 to the per-stack comparisons, I think that might be important enough to have that as part of the initial reminder logic rather than a future improvement.

**Comment by krysal** [comment]
I prefer to have the PR limit as global rather than apply it per stack, perhaps just excluding the infrastructure repo, given it changes more irregularly, at least as an initial experiment while we refine the planning and issue assignation process.

The reviewers overlap quite a bit so I don't think it helps to allow more PRs across the stack (unless previously planned) for the burden of increased the number and the context switching it implies. This can also help folks who have expressed difficulty focusing on a project or given tasks in the past.

**Comment by sarayourfriend** [comment]
Agreed, Krystle. The stack-weighted approach would require a lot more fiddly maths to accurately tell what the "threshold" should be.

**Comment by obulat** [comment]
Thank you for the detailed calculations, @sarayourfriend. The fact that three open PRs per maintainer means 7 PRs for review is really mind-blowing. There are also PRs opened by community contributors.

One note I have about the count is related to project planning. I think we should not include the project proposals and implementation plans from this count as they were suggested as "alternative work" elsewhere.

**Comment by sarayourfriend** [comment]
Totally. @krysal suggested we exclude certain labels from the count and @zackkrida has added that in the associated PR: #2303

It's an excellent improvement, thank you both for the thoughtfulness with that.
